tests/test_diffraction_objects.py

```diff
@@ -212,21 +213,41 @@ def _test_valid_diffraction_objects(actual_diffraction_object, function, expecte
 
 
 def test_get_angle_index():
-    test = DiffractionObject()
-    test.angles = np.array([10, 20, 30, 40, 50, 60])
-    actual_angle_index = test.get_angle_index(angle=10)
-    assert actual_angle_index == 0
+    test = DiffractionObject(
+        wavelength=0.71, xarray=np.array([30, 60, 90]), yarray=np.array([1, 2, 3]), xtype="tth"
+    )
+    actual_index = test.get_array_index(xtype="tth", value=30)
+    assert actual_index == 0
+
+
+params_index_bad = [
+    # UC1: empty array
+    (
+        [0.71, np.array([]), np.array([]), "tth", "tth", 10],
+        [IndexError, "WARNING: no matching value 10 found in the tth array."],
```

Comment:
is this an error or a warning?

Reply:
Should be an error. I've edited the error message.